moto/sns/models.py

```diff
@@ -388,11 +388,21 @@ def update_sms_attributes(self, attrs):
         self.sms_attributes.update(attrs)
 
     def create_topic(self, name, attributes=None, tags=None):
-        fails_constraints = not re.match(r"^[a-zA-Z0-9_-]{1,256}$", name)
+
+        if (
+            attributes.get("FifoTopic")
+            and attributes.get("FifoTopic").lower() == "true"
+        ):
+            fails_constraints = not re.match(r"^[a-zA-Z0-9_.-]{1,256}fifo$", name)
```

Comment:
Need to move the `.` inside the regex:
`r"^[a-zA-Z0-9_-]{1,256}\.fifo$"`

Should have said this earlier, but worth creating another test case that verifies that `queue.name.fifo` also throws an error. (That name would be considered valid with the current regex, but shouldn't be.)